garak/generators/base.py

```diff
@@ -127,7 +124,7 @@ def generate(self, prompt: str, generations_this_call: int = -1) -> List[str]:
                 )
                 generation_iterator.set_description(self.fullname[:55])
                 for i in generation_iterator:
-                    outputs.append(self._call_model(prompt, generations_this_call))
+                    outputs.append(self._call_model(prompt, generations_this_call=1))
```

Comment:
This is an interesting change, `generations_this_call=1` will set the local variable to `1`, is that the desired action? Consider:

```suggestion
                    outputs.append(self._call_model(prompt, 1)) # generate once as `generation_iterator` consumes `generations_this_call`
```

The comment clarifies the reasoning.

Reply:
Oh, thanks, that was more subtle. Changing the local variable was indeed not the desired action.

The bug this fixes came from a lack of clarity around process and expectations in generators; created #682 to prompt documentation for this.